### PR TITLE
Upstream Grenades: HESH Rocket, Buffed Fragmentation, Pipebomb, WP & Fix

### PR DESCRIPTION
**Pull request by WeNeedMorePhoron**
## About The Pull Request
- Port in most of #7850 Upstream, White Phosphorus Grenades (mispelled in code)
- Port in most of #7025 - Adding HESH Rocket, Buffed Fragmentation, Stinger Grenades. Modular Occulus Stinger code & sprites have been replaced with theirs with name override. Double check if the names I've made up for the unused FTU grenade is fine. (They added it but haven't used it upstream yet)
- Port in #7187 - Added blast radius to frag grenades
- Ported in the three pipebombs PRs. Forgot name. It can be crafted in crafting menu.
- Ported in two QDEL fix from above in an attempt to fix the projectile duping issue. Improved ricochet message.
- Cleaned up the horrible projectile loop

**ORIGINAL CHANGES**
- Due to WP Grenades being extremely underwhelming (taking 3 - 6 minutes to knock someone down), I've buffed fire stack by roughly 2x by making them cause additional burn damage and make it less ignorable. Getting full fire stacked without clothes should knock you down in around 1 minute or so.
- WP Grenades are accessible to traitor through uplink (Unused except in centcomm upstream)
- Fixed a server crash / duping issue with grenade launcher. This will be PRed upstream if I am able to replicate it upstream.
- If an item deal less than 5 damage, they won't knock you down when they fall from above. This usually corresponds to tiny and small item. Normal item will knock you down. An additional red text damage is shown if an item knock you down.

## Why It's Good For The Game
fix crashes
i love explosives
hate authority

## Changelog
```changelog
add: Craftable Pipebomb made with plasma cell 
add: HESH Rockets. RPG-7 package starts with 4 out of 12 rockets being HESH. HESH rocket have worse blast but send a lot of fragments to the other side of a room.
add: White phosphorus grenades, accessible to traitor for now.
tweak: Normal HE Rocket nerfed a bit, upstream parity. Still kill with a direct hit. 
tweak: Stinger Grenades sprite changed to match upstream, fragmentation grenades now cause a 2 tiles small blast around itself. 
tweak: Grenades dropping on someone from above should now bonk off someone's head.
tweak: Grenade fragments now ricochet making them much deadlier, especially in small room.
tweak: Grenade now generally produce 1/3 the fragment but deals 3x the damage. Upstream parity, slight buff, frag grenades and pipebomb has a blast radius of 2x. Grenade deadliness against unarmored people is practically unchanged, grenade deadliness against CAS Voidsuit / Deepflare goes from 0 damage to 110 at 0 - 1 tile range, rapidly dropping off to 11 - 30 at 2 - 3 tiles range.
tweak: Fire stacks are much deadlier, approximately by 2x, making being on fire less ignorable. They also put themselves out slowly.
tweak: Falling item that deals less than 5 damage won't knock you down anymore (Tiny and Small Item)
fix: A crash that can happen with grenade launcher launched grenade.
```

### CODE REVIEW

**Comment by WeNeedMorePhoron** [comment]
After writing out the PR I realized I forgot to check the bonking feature (Because that require a multikey). Please hold off on merging until I confirm it work.

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -441,7 +441,7 @@
 		pixel_x = location.pixel_x
 		pixel_y = location.pixel_y
 
-		if(!bumped && !isturf(original))
+		if(!bumped && !QDELETED(original) && !isturf(original))
```
Comment (WeNeedMorePhoron) [inline code/modules/projectiles/projectile.dm:444]: upstream change

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -418,7 +418,7 @@
 /obj/item/projectile/Process()
 	var/first_step = TRUE
 
-	spawn while(src && src.loc)
+	while(!QDELETED(src) && src.loc)
```
Comment (WeNeedMorePhoron) [inline code/modules/projectiles/projectile.dm]: loop was utterly cursed so i changed it

**Comment by WeNeedMorePhoron** [comment]
Ready for review

**Review by Jamini — CHANGES_REQUESTED** [review]
Please test all of these and give us rough damage values.

We had to nerf several explosions from Eris because they were onetaps.

**Review by Shadow-Quill — COMMENTED** [review]
No complaints code-wise.

**Review by Jamini — APPROVED** [review]

**Pull request merged**